packages/template/src/Adaptor.js

```diff
@@ -2,9 +2,57 @@ import {
   execute as commonExecute,
   composeNextState,
 } from '@openfn/language-common';
-import { expandReferences } from '@openfn/language-common/util'
-import { request } from './Utils';
 
+import {
+  request as commonRequest,
+  expandReferences,
+} from '@openfn/language-common/util';
+
+import { makeBasicAuth, assertConfiguration } from './Utils';
+
+export function request(method, path, params, callback) {
+  return state => {
+    const [resolvedPath, resolvedParams] = expandReferences(
+      state,
+      path,
+      params
+    );
+
+    const { username, password, baseUrl } = assertConfiguration(
+      state.configuration
+    );
+    // Initialize headers as an empty object
+    let headers = {};
+    // Check if resolvedParams has headers, and merge them into the headers object
+    if (resolvedParams?.headers) {
+      headers = resolvedParams.headers;
+    }
+    // Set the 'content-type' header if it's not already set
+    if (!headers['content-type']) {
+      headers['content-type'] = 'application/json';
+    }
+
+    headers['Authorization'] = makeBasicAuth(username, password);
+
+    const options = {
+      ...resolvedParams,
+      headers,
+      baseUrl,
+    };
+
+    return commonRequest(method, resolvedPath, options)
+      .then(response => {
+        const { method, url, body, code, duration } = response;
+        console.log(method, url, '-', code, 'in', duration + 'ms');
```

Comment:
This is REALLY interesting.

This is painfully boring and rote code and I do not want adaptor implementors to have to write it.

But we're making them, right, because the underlying common helper doesn't auto log. We're doing it manually in http and we're doing it manually here.

This is just an idea, but what if the response had some kind of `log()` function built into it, which logs the message printed by HTTP. Now any adaptor can easily and painlessly print out a generic HTTP response log.

Actually, we should also consider that most adaptors don't just want to log the http details. Most adaptors would say "Created patient bob (222ms)" and ignore the http stuff.

Reply:
I like this idea, Can i suggest we have `response.data()` and `response.log()` then, It looks more organized 

```
// The new response structure 
return {
  data: () => ({
    method,
    url: responseUrl(fullUrlOrPath, response),
    code: response.statusCode,
    headers: response.headers,
    body: responseBody,
    message: statusText,
    duration,
  }),
  log: (message) => {
    if (message) {
      console.log(message);
    } else {
      console.log(
        method,
        fullUrlOrPath,
        "-",
        response.statusCode,
        "in",
        duration + "ms"
      );
    }
  },
};

```

Reply:
```
// response.log()
log: () => console.log(
        method,
        fullUrlOrPath,
        "-",
        response.statusCode,
        "in",
        duration + "ms"
      );```